x/stake/handler.go

```diff
@@ -184,31 +148,31 @@ func delegate(ctx sdk.Context, k Keeper, delegatorAddr sdk.Address,
 	if err != nil {
 		return nil, err
 	}
-	pool, candidate, newShares := pool.candidateAddTokens(candidate, bondAmt.Amount)
+	validator, pool, newShares := validator.addTokensFromDel(pool, bondAmt.Amount)
 	bond.Shares = bond.Shares.Add(newShares)
 
 	// Update bond height
 	bond.Height = ctx.BlockHeight()
 
-	k.setDelegatorBond(ctx, bond)
-	k.setCandidate(ctx, candidate)
+	k.setDelegation(ctx, bond)
+	k.setValidator(ctx, validator)
 	k.setPool(ctx, pool)
-	tags := sdk.NewTags("action", []byte("delegate"), "delegator", delegatorAddr.Bytes(), "candidate", candidate.Address.Bytes())
+	tags := sdk.NewTags("action", []byte("delegate"), "delegator", delegatorAddr.Bytes(), "validator", validator.Address.Bytes())
 	return tags, nil
 }
 
 func handleMsgUnbond(ctx sdk.Context, msg MsgUnbond, k Keeper) sdk.Result {
 
 	// check if bond has any shares in it unbond
-	bond, found := k.GetDelegatorBond(ctx, msg.DelegatorAddr, msg.CandidateAddr)
+	bond, found := k.GetDelegation(ctx, msg.DelegatorAddr, msg.ValidatorAddr)
 	if !found {
 		return ErrNoDelegatorForAddress(k.codespace).Result()
 	}
 	if !bond.Shares.GT(sdk.ZeroRat()) { // bond shares < msg shares
```

Comment:
shouldnt this be bond.Shares.GT(msg.Shares), not just 0 ?

Reply:
actually an error that check was redundant to the real check which happens later. Deleted